packages/react-core/src/components/TreeView/TreeViewListItem.tsx

```diff
@@ -186,10 +186,10 @@ export const TreeViewListItem: React.FunctionComponent<TreeViewListItemProps> =
               )}
               onClick={(evt: React.MouseEvent) => {
                 if (!hasCheck) {
-                  if (children) {
+                  onSelect && onSelect(evt, itemData, parentItem);
+                  if (children && evt.isDefaultPrevented() !== true) {
```

Comment:
If I'm following this correctly, expansion would be prevented if `evt.preventDefault()` is called within the user-defined `onSelect` right? If so, then could you please update the `onSelect` property description to reflect the new behavior interaction?